lib/mayaHydra/hydraExtensions/mhLeadObjectPathTracker.cpp

```diff
@@ -65,6 +65,15 @@ class GlobalSelectionChangedObs : public Ufe::Observer
             MayaHydra::MhLeadObjectPathTracker& _leadObjectPathTracker;
     };
 
+void pushHydraSelectionsToPathVector(const Fvp::PrimSelectionInfoVector& hydraSelections, SdfPathVector& outPathVector)
```

Comment:
This doesn't seem to be used to append to outPathVector, which is always empty, as far as I can tell.  Would be clearer (and allow for const safeness) if you just returned the SdfPathVector.  Return value optimization or move construction will take care of efficiency.
```suggestion
SdfPathVector hydraSelectionsPaths(const Fvp::PrimSelectionInfoVector& hydraSelections)
```

Reply:
Removed it and used the newly created `SceneIndexPaths` method instead